src/appmixer/utils/controls/OnError/OnError.js

```diff
@@ -1,59 +1,94 @@
-        }
-    });
+async function fetchErrorsPage(context, lock, limit, logId = null, lastGridTimestamp = null, size = 100, errors = []) {
+
+    const qs = {
+        size: Math.min(limit, size),
+        query: 'severity:error',
+        flowId: context.flowId
+    };
+
+    if (logId && lastGridTimestamp) {
+        qs.searchAfter = [logId, lastGridTimestamp];
+    }
+
+    while (true) {
+        try {
+            const result = await context.callAppmixer({
+                endPoint: '/logs',
+                method: 'GET',
+                qs
+            });
+
+            if (!(result.hits && result.hits.length > 0 && limit > result.hits.length)) {
+                break;
+            }
 
-    errors = errors.concat(result.hits);
+            lock.extend(parseInt(context.config.lockTTL, 10) || 1000 * 60 * 2);
+            const lastLog = result.hits[result.hits.length - 1];
+            errors.push(...result.hits);
 
-    if (result.hits.length === 0 || result.hits.length < size) {
-        return errors;
+            if (errors.length >= limit) {
+                break;
+            }
+
+            qs.searchAfter = [lastLog['_id'], lastLog['gridTimestamp']];
+        } catch (error) {
+            console.error('Error fetching errors:', error);
+            await context.log({ level: 'error', error: error.message || error });
+            break;
+        }
     }
 
-    return getErrors(context, now, errors, result.nextFrom, size);
+    return errors;
 }
 
-/**
- * Add Flow name and component labels to those errors.
- * @param {Context} context
- * @param {Array} errors
- */
+// Adds labels to errors for better identification
 function addLabels(context, errors) {
 
-    return errors.map(err => {
-        return Object.assign(
-            err,
-            {
-                flowName: context.flowInfo.flowName
-            },
-            err.componentId ? {
-                componentLabel: context.flowDescriptor[err.componentId].label || err.componentType.split('.')[3]
-            } : {});
-    });
+    return errors.map(err => ({
+        ...err,
+        flowName: context.flowInfo.flowName,
+        componentLabel: err.componentId ? (context.flowDescriptor[err.componentId]?.label || err.componentType.split('.')[3]) : undefined
+    }));
 }
 
 module.exports = {
 
-    async start(context) {
+    async tick(context) {
+
+        const lockName = `errorProcessingLock-${context.componentId}`; // Define a unique lock name
+        let lock;
+        try {
+            lock = await context.lock(lockName, { ttl: context.config.lockTTL || 1000 * 60 * 5, maxRetryCount: 0 });
+            const { lastLogId, gridTimestamp } = context.state;
+            // ensure that if there is a limit set by context.config.limit
+            const limit = context.properties.limit
+                ? Math.min(
+                    parseInt(context.properties.limit),
+                    parseInt(context.config.limit) || Infinity
```

Comment:
So there is `Infinity` limit? if not in `context.config.imit`? That is not good. In a combination with that `while(true)` you can get to a situation where you load tons of errors. The engine can easily generate tens of thousands of errors and you don't want to get all of them.
I would set the `default` (if no `context.config.limit`) to something like 1000.